lib/wrappers/openssl.nim

```diff
@@ -25,8 +25,14 @@ when useWinVersion:
 
   from winlean import SocketHandle
 else:
-  const
-    versions = "(|.38|.39|.41|.43|.10|.1.0.2|.1.0.1|.1.0.0|.0.9.9|.0.9.8)"
+  when defined(openssl_before_1_1):
+    # Avoid explicitely loading OpenSSL 1.1.0
+    # If no other named versions are found it might still
+    # loaded from an unversioned ".so" file
+    const versions = "(.38|.39|.41|.43|.10|.1.0.2|.1.0.1|.1.0.0|.0.9.9|.0.9.8|)"
+  else:
+    const versions = "(.1.1|.38|.39|.41|.43|.10|.1.0.2|.1.0.1|.1.0.0|.0.9.9|.0.9.8|)"
```

Comment:
Hrm, I thought the idea was to prevent the loading of pre-1.1.0 openssl versions if ``openssl_before_1_1`` is not defined? This just changes the order in which these are loaded, is this intended?

Reply:
The current code is doing what you described in the 6 cases above, with the only exception of being able to use OpenSSL < 1.1.0 even when "-d:openssl_before_1_1" is **not** passed. This provides backward compatibility.
If there's a strong reason to error out I'll do that, but at this time the only "price" you pay for "mismatching" the version is losing some methods.
(Newer methods e.g. SSL cert check are not implemented in this PR. )

Reply:
All this define does is change the order that the DLLs are looked up in. I don't see the point in doing this. Why not just always look for the newest first?

Reply:
The behavior around initialization and SSL certificate checks is different. If the user decided to explicitly pass openssl_before_1_1 for some reason we should try to use an older version if possible.

Reply:
Sorry, I'm not entirely sure what you mean. I still don't understand why you would ever want to load an older version.